packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -298,6 +298,7 @@ export abstract class RestApiBase extends Resource implements IRestApi {
    */
   public deploymentStage!: Stage;
 
+  protected resourcePolicyDocument?: iam.PolicyDocument;
```

Comment:
Mark this as `@internal`.

```suggestion
  /** @internal */
  protected resourcePolicyDocument?: iam.PolicyDocument;
```